### PR TITLE
Add string interpolation support for `W_str` and `W_cmd`

### PR DESCRIPTION
**Pull request by hammerfunctor**

### CODE REVIEW

File: src/eval.jl [hunk 1]
```diff
@@ -65,5 +65,5 @@ function parseexpr(str::AbstractString)
 end
 
 macro W_cmd(str)
-    parseexpr(str)
+    quote parseexpr($(esc(Meta.parse("\"$(escape_string(str))\"")))) end
```
Comment (simonbyrne) [inline src/eval.jl:68]: I'm not sure this will work: it's not clear that Wolfram code is valid Julia code.

File: src/types.jl [hunk 2]
```diff
@@ -37,7 +37,7 @@ Base.show(io::IO, s::WSymbol) = print(io, 'W', '"', s.name, '"')
 Base.:(==)(a::WSymbol, b::WSymbol) = a.name == b.name
 
 macro W_str(str)
-    WSymbol(str)
+    quote WSymbol($(esc(Meta.parse("\"$(escape_string(str))\"")))) end
```
Comment (simonbyrne) [inline src/types.jl:40]: This is wrong: `W"name"` should just return a symbol

File: src/eval.jl (hunk 1 above)
Comment (hammerfunctor) [reply src/eval.jl:68]: It has nothing to do with Wolfram code parsing. It just inspect a LITERAL string and all expand `$` interpolated variables

File: src/types.jl (hunk 2 above)
Comment (hammerfunctor) [reply src/types.jl:40]: It does.
```julia
julia> import Pkg; Pkg.activate("."); using MathLink
  Activating project at `~/dev/sw/MathLink.jl`

julia> let e=W`Sin[Cos[$(log(2))]]`
           println(e)
           println(weval(e))
       end
W"Sin"(W"Cos"(0.6931471805599453))
0.6955886362231636

julia> let e=W"name"
           println(e)
           println(typeof(e))
           println(weval(W"Head"(e)))
       end
name
MathLink.WSymbol
Symbol
```

File: src/eval.jl (hunk 1 above)
Comment (simonbyrne) [reply src/eval.jl:68]: Ah, that makes sense. Would be good to add some tests.

File: src/types.jl (hunk 2 above)
Comment (simonbyrne) [reply src/types.jl:40]: Ah, got it.

**Review by hammerfunctor — COMMENTED** [review]
tests added

**Comment by hammerfunctor** [comment]
Dude, you will merge this PR, right? @simonbyrne

**Pull request merged**